runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2499,7 +2546,23 @@ bool TR::CompilationInfo::shouldRetryCompilation(TR_MethodToBeCompiled *entry, T
    // don't carry the decision to generate AOT code to the next compilation
    // because it may no longer be the right thing to do at that point
    if (tryCompilingAgain)
+      {
       entry->_useAotCompilation = false;
+      }
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+   else if (entry->_compErrCode != compilationOK)
+      {
+      TR::CompilationInfo *compInfo = entry->_compInfoPT->getCompilationInfo();
+      J9JavaVM *javaVM = compInfo->getJITConfig()->javaVM;
+      J9VMThread *vmThread = javaVM->internalVMFunctions->currentVMThread(javaVM);
```

Comment:
Looks like `_compInfoPT` is of type `TR::CompilationInfoPerThreadBase` which doesn't have the `getCompilationThread()` method, so it might be ugly to explicitly cast to a `TR::CompilationInfoPerThread` (also not sure if it's always correct given JITServer). However, given that `shouldRetryCompilation` is only called in two places, might just be cleanest to pass in the vmThread as a parameter.